### PR TITLE
Various libClusterFuzz improvements.

### PR DESCRIPTION
**Pull request by oliverchang**
- Add a context manager for setting necessary env vars temporarily.
- Set exit codes for fuzz and reproduce.
- Allow additional args to be passed to fuzzing.
- Create prune output directory if needed.

For https://github.com/google/oss-fuzz/pull/6375

### CODE REVIEW

File: src/clusterfuzz/environment/__init__.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+# limitations under the License.
+"""Environment helpers."""
+
+import contextlib
+import os
+
+from clusterfuzz._internal.system import environment
+
+# Inverted SANITIZER_NAME_MAP.
+SANITIZER_MAP = {v: k for k, v in environment.SANITIZER_NAME_MAP.items()}
```
Comment (jonathanmetzman) [inline src/clusterfuzz/environment/__init__.py:22]: Should we assert that each value in SANITIZER_NAME_MAP is unique?

File: src/clusterfuzz/fuzz/__main__.py [hunk 2]
```diff
@@ -47,28 +46,28 @@ def main():
...
-    options.merge_back_new_testcases = False
-    options.analyze_dictionary = False
+    result = engine_impl.fuzz(args.target, options, args.output,
+                              args.max_duration)
 
-  result = engine_impl.fuzz(args.target, options, args.output,
-                            args.max_duration)
   print('Command: ', ' '.join([shlex.quote(part) for part in result.command]))
+  if result.crashes:
+    sys.exit(1)
```
Comment (jonathanmetzman) [inline src/clusterfuzz/fuzz/__main__.py:68]: I'd prefer we returned these values and that `sys.exit(main())` was called on line 74

File: src/clusterfuzz/environment/__init__.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    if self.interactive:
+      self.set_value('CF_INTERACTIVE', 'True')
+
+    return self
+
+  def __exit__(self, *exc):
+    """Exit environment."""
+    del exc
+
+    for key, value in self._previous_values.items():
```
Comment (jonathanmetzman) [inline src/clusterfuzz/environment/__init__.py:56]: We're sure that none of these will get deleted?
Maybe check for key presence instead just to be safe?

**Review by jonathanmetzman — APPROVED** [review]
LGTM

File: src/clusterfuzz/environment/__init__.py (hunk 1 above)
Comment (oliverchang) [reply src/clusterfuzz/environment/__init__.py:22]: Good point, added!

File: src/clusterfuzz/fuzz/__main__.py (hunk 2 above)
Comment (oliverchang) [reply src/clusterfuzz/fuzz/__main__.py:68]: That's a much bigger change, and not possible today without a lot of refactoring.

File: src/clusterfuzz/environment/__init__.py (hunk 3 above)
Comment (oliverchang) [reply src/clusterfuzz/environment/__init__.py:58]: Could you clarify what you mean here? 

We only delete an env var if we previously got "None" when we saved it.

**Pull request merged**